core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -13,31 +13,829 @@
+    skillName: string,
+    questionCount: number
+  ): Promise<void> {
+    for (let i = 0; i < questionCount; i++) {
+      await this.addBasicAlgebraQuestionToSkill(skillName);
+    }
+  }
+
+  /**
+   * Create a basic algebra question in the skill editor page.
+   */
+  async addBasicAlgebraQuestionToSkill(skillName: string): Promise<void> {
+    await this.openSkillEditor(skillName);
+    await this.clickOn(createQuestionButton);
+    await this.clickOn(textStateEditSelector);
+    await this.page.waitForSelector(richTextAreaField, {visible: true});
+    await this.type(richTextAreaField, 'Add 1+2');
+    await this.page.waitForSelector(`${saveContentButton}:not([disabled])`);
+    await this.clickOn(saveContentButton);
+
+    await this.clickOn(addInteractionButton);
+    await this.page.waitForSelector(interactionNumberInputButton, {
+      visible: true,
+    });
+    await this.clickOn(interactionNumberInputButton);
+    await this.clickOn(saveInteractionButton);
+    await this.page.waitForSelector('oppia-add-answer-group-modal-component', {
+      visible: true,
+    });
+    await this.clickOn(responseRuleDropdown);
+    await this.clickOn(equalsRuleButtonText);
+    await this.type(floatTextField, '3');
+    await this.clickOn(answersInGroupAreCorrectToggle);
+    await this.clickOn(saveResponseButton);
+    await this.page.waitForSelector(modalDiv, {hidden: true});
+
+    await this.clickOn(defaultFeedbackTab);
+    await this.clickOn(openOutcomeFeedBackEditor);
+    await this.clickOn(richTextAreaField);
+    await this.type(richTextAreaField, 'The answer is 3');
+    await this.clickOn(saveOutcomeFeedbackButton);
+
+    await this.clickOn(addHintButton);
+    await this.page.waitForSelector(modalDiv, {visible: true});
+    await this.type(richTextAreaField, '3');
+    await this.clickOn(saveHintButton);
+    await this.page.waitForSelector(modalDiv, {hidden: true});
+
+    await this.clickOn(addSolutionButton);
+    await this.page.waitForSelector(modalDiv, {visible: true});
+    await this.page.waitForSelector(answerTypeDropdown);
+    await this.page.select(answerTypeDropdown, 'The only');
+    await this.page.waitForSelector(solutionFloatTextField);
+    await this.type(solutionFloatTextField, '3');
+    await this.page.waitForSelector(`${submitAnswerButton}:not([disabled])`);
+    await this.clickOn(submitAnswerButton);
+    await this.type(richTextAreaField, '1+2 is 3');
+    await this.page.waitForSelector(`${submitSolutionButton}:not([disabled])`);
+    await this.clickOn(submitSolutionButton);
+    await this.page.waitForSelector(modalDiv, {hidden: true});
+
+    await this.clickOn(saveQuestionButton);
+  }
+
+  /**
+   * Navigate to the creator dashboard page.
+   */
+  async navigateToCreatorDashboardPage(): Promise<void> {
+    await this.page.bringToFront();
+    await this.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Create an exploration as a curriculum admin.
+   */
+  async createAndPublishExploration(
```

Comment:
Same here, I would expect this to be in the exploration editor utils, unless I'm missing some context.

Reply:
We're testing the curriculum admin role here though - not the exploration editor. Curriculum admins can also create and publish explorations. Ditto above, seems to be recently pushed code.

Reply:
Sorry to clarify AFAIK we are trying to separate user behaviors. Yes a curriculum admin can create and publish explorations but so can every user, we wouldn't want to duplicate this for every user role. We use composition which allows a user to have multiple classes. ATM the exploration editor class is composed by default so any function you put in there can be ran by any created user.

Reply:
Unless this function has something specific to the curriculum admin role we should separate it into the exploration editor class.

Reply:
Ah okay, thanks for explaining, so are you suggesting to create a separate exploration editor user in the spec to deal with exploration creation if I'm not misunderstanding? I didn't quite get what you meant by a user having multiple classes.

Reply:
No it's fine to have one user just move these functions over to the other file and concat the types.

Reply:
E.g. when we you declare curriculumAdmin: CurriculumAdmin & ExplorationEditor

Reply:
Done, thanks!